### ISSUE TITLE
JsonAssertions expects non null actual parameter

### ISSUE THREAD

**Issue by filiphr** [lukas-krecan/JsonUnit#298]
There is a warning in IntelliJ when trying to use `assertThatJson` with a potentially `null` actual value.
In order to avoid the exception I have to write

```
JsonNode node = functionThatCanReturnNull();
assertThat(node).isNotNull();
asserThatJson(node).isEqualTo("...");
```

I shouldn't need to do an explicit null assertion in my tests.
I would expect that this is entirely handled by the library itself.

Is there a reason why you can't throw an assertion failure if you try to create `ConfigurableJsonAssert` with a `null` actual value?

**Comment by lukas-krecan** [comment]
HI, thanks for feedback. I will take a look at it. Most likely it's only incorrect nullability annotation.

**Comment by filiphr** [comment]
Thanks, the `nullability` you might be right. I scrolled a bit through the API and it seems that an NPE that was there before is not possible. 

I remember that in some version there was an NPE when you passed `null`.

**Comment by lukas-krecan** [comment]
Currently, you get `java.lang.NullPointerException: Can not make assertions about null JSON.` if you pass in null, so the annotation and IntelliJ warning is correct. But I understand your use case, maybe the annotation should change after all. Will think about it.

**Comment by filiphr** [comment]
Thanks for looking into it.

In my opinion the whole point in having an assertion framework is to catch these things. If I am saying that the actual should be equal to something, but it isn't then it should fail the assertion. Perhaps instead of thrown an NPE you can throw an assertion failure

**Comment by lukas-krecan** [comment]
Changed the bahavior, now `null` is permitted as actual value, will release it tomorrow. Thanks again for feedback.

**Comment by filiphr** [comment]
Thanks, this is much appreciated.

**Comment by lukas-krecan** [comment]
Released as 2.20.0
